### PR TITLE
Allows Xenos to pat ground fire

### PR DESCRIPTION
**Pull request by Shad0vvs**
## About The Pull Request

Allows xenos to pat fire on the ground, better than sitting at a flank waiting for it to go away I suppose, worth a TM.
Theoretic numbers: 4 pats for normal fire, 5 pats for green fire, 8 pats for blue fire.

## Why It's Good For The Game

Current number is fairly weak, can be adjusted easily later.

## Changelog

:cl:
balance: Allows Xenos to pat out fires on the ground while on help intent.
/:cl:

### CODE REVIEW

**Comment by c4xmaniac** [comment]
lore reason is the xenos are vomiting acid on to the fire, its the vomit fire brigade.

**Comment by usnpeepoo** [comment]
Ok now make it so their next slash does burn damage (patting fire heats their claws)

**Comment by Diegoflores31** [comment]
is there a exception to white fire ? i mean they should not be able to put out those

**Comment by Shad0vvs** [comment]
> is there a exception to white fire ? i mean they should not be able to put out those

Not currently no, but it does take a lot of pats to put out. Could be added later if the need arises.

**Comment by Wintermote** [comment]
This is making an already weak weapon even weaker

I'm not exactly against this but we really need fire tiles to be something xenos fear and not something they will casually ignore/run through to get grabs and kills with.

**Comment by stanalbatross** [comment]
> This is making an already weak weapon even weaker
> 
> I'm not exactly against this but we really need fire tiles to be something xenos fear and not something they will casually ignore/run through to get grabs and kills with.

what are you on bro, fire is incredibly strong rn especially greenfire

**Comment by gohdatrice** [comment]
> I'm not exactly against this but we really need fire tiles to be something xenos fear and not something they will casually ignore/run through to get grabs and kills with.

Right now xenos don't "fear" fire tiles. They find them tedious because you can't do anything but just sit there and wait for them to run out (at which point marines will probably reflame them anyway). Not that that's entirely bad, it can force xenos find flanks which can be interesting. But xenos do not and never will "fear" lingering fire tiles.
Now the actual flamer itself, xenos do fear. Especially on the backlines as a runner, if you get set on fire you're probably dead. But even on the front getting set on fire can be pretty bad. But that's to do with getting hit by the flamer itself, not the lingering tiles. Letting xenos pat the tiles wouldn't change that.

Basically in my mind the main thing this PR changes is whether or not xenos are allowed to try pushing through a choke that marines are flaming, or if they have to just sit there waiting/find a flank. It also affects xenos that have their escape route cut off by fire, as now they can in theory put it out (assuming they don't get gunned down first, 4 pats is still a lot if you're trying to escape)

**Comment by Don2137** [comment]
Boohoo I can't poopshit the entire frontline in flames anymore

Good change

**Comment by Lafrien** [comment]
Backliners can now pat out the lucky flame that was cutting off Queen's push making a safe way of retreat. Maybe increase the lingered flame damage as well just a little bit so that there was some balance instead of straight buff to Xenos? Also is there any cooldown difference for patting, like will runner pat it out faster, or same "time" (not ammount of pats, the speed) as drone?

**Comment by Blithering** [comment]
Ok, but marines should be able to waft away boiler gas for the same reasons.

**Review by stanalbatross — APPROVED** [review]

**Comment by GoldenDarkness55** [comment]
Remember snow? Don't you love clicking snow tiles constantly? Now you can do the same with fire! - At least make it a timed action
I dunno why this is even needed though - just modify flame linger duration if it's too restrictive, there's really no reason for fire to last THAT long, especially blue.

That aside, this is just a plain bad idea as it makes fire useless at the thing it's been designed for - preventing xenos from running through it and buying time. Blue flaming a pathway only for 2 roonies to remove it in 3 seconds? Shouldn't acid extinguishing power be buffed alongside this, or perhaps instead?

**Comment by Shad0vvs** [comment]
> Remember snow? Don't you love clicking snow tiles constantly? Now you can do the same with fire! - At least make it a timed action I dunno why this is even needed though - just modify flame linger duration if it's too restrictive, there's really no reason for fire to last THAT long, especially blue.
> 
> That aside, this is just a plain bad idea as it makes fire useless at the thing it's been designed for - preventing xenos from running through it and buying time. Blue flaming a pathway only for 2 roonies to remove it in 3 seconds? Shouldn't acid extinguishing power be buffed alongside this, or perhaps instead?

Just don't let them click the fire.

**Comment by Lumdor** [comment]
Instead of fixing the actual issues with fire let's make it so players need to interact with fire the same way they do with snow tiles.

**Comment by usnpeepoo** [comment]
i'll be honest greenfires issues cannot be fixed by simple patting. it's a part of the issue, but this PR doesn't fix it completely.  

The best example of this is greenfire stacking during fob sieges. Not only can you not approach it (to pat it), nor can you even withstand it, the effects it gives (slowdown, armor removal) makes people shooting at the cades have an extremely easy target to shoot which is also debuffed. 
Another example is throwing an incin (or using IFF GL) behind a xeno on their only escape. Normally, with any other type of fire they simply get set on fire allowing you to chase. With greenfire this isn't even an option. The xeno simply dies on the spot due to being forced to walk through 3-5 tiles of flame that slows them and removes their armor.
While this PR helps with t3's being forced to play observer when an entire front is greenfired, it doesn't help with any of the issued state above, which is still a major problem with current hvx balance.

**Comment by FoxyStalin** [comment]
Nerfing all fire to fix greenfire's broken aspects isn't a good idea, because within the month Greenfire is going to be nerfed and then all fire will be useless instead.

If you think Greenfire is too strong right now, nerf it specifically.  Make it do slowdown or armor removal, not both, it's the most obvious fix in the history of fixes.  
And another thing,  why are xenos getting buffs to make them entierly self-sufficient, yet marines are expected to run everything through a chain of interactions.  Every xeno got the ability to slash rail lights because they can't be assed to get one of the 5 castes with acid to come by and melt them,  and now this?   Lets give runners and lurkers weeding while we're at it.

**Comment by Shad0vvs** [comment]
Hello,

This PR is unrelated to greenfire.

Thanks!

**Comment by FoxyStalin** [comment]
> Hello,
> 
> This PR is unrelated to greenfire.
> 
> Thanks!

So you're nerfing fire on a whim then?  That's not really better.

**Comment by Shad0vvs** [comment]
> > Hello,
> > This PR is unrelated to greenfire.
> > Thanks!
> 
> So you're nerfing fire on a whim then? That's not really better.
If not being able to fire a location then walk away from it and leave it unguarded means it is nurfed, then yes in theory.

**Comment by FoxyStalin** [comment]
> > So you're nerfing fire on a whim then? That's not really better.
> 
> If not being able to fire a location then walk away from it and leave it unguarded means it is nurfed, then yes in theory.

The thing being xenos can already do this.  Any xeno with acid spray can eliminate a firewall that's blocking a flank. This is just removing the teamwork from it and allowing already self-sufficient backline castes to be even more self-sufficient.  
If you don't see how giving every xeno the ability to bypass fire in 3-5 clicks isn't a nerf to fire,  I don't really know what to tell you.

**Comment by Shad0vvs** [comment]
> > > So you're nerfing fire on a whim then? That's not really better.
> > 
> > 
> > If not being able to fire a location then walk away from it and leave it unguarded means it is nurfed, then yes in theory.
> 
> The thing being xenos can already do this. Any xeno with acid spray can eliminate a firewall that's blocking a flank. This is just removing the teamwork from it and allowing already self-sufficient backline castes to be even more self-sufficient. If you don't see how giving every xeno the ability to bypass fire in 3-5 clicks isn't a nerf to fire, I don't really know what to tell you.

Imagine the teamwork when a couple Xenos come together on a flank and pat out the fire and preform said flank!

**Comment by ApplePlastic** [comment]
"ah yes just let me pat this thousand degree fire with no consequences whatsoever"

This makes fire less of an area-of-denial and more of a crutch for xenos, which.. Entirely defeats the purpose of fire. Xenos can already spit on multiple tiles of fire to extinguish it, which costs some plasma to do. They don't need to be able to extinguish fire without consequence.

**Comment by Don2137** [comment]
Dudes in here need a reality check. A flamer can shit out two lines in the time it will take a xeno to pat out a single tile lol.

> Xenos can already spit on multiple tiles of fire to extinguish it, which costs some plasma to do. They don't need to be able to extinguish fire without consequence.

Please actually play xeno before commenting on things you have no idea about

**Comment by ApplePlastic** [comment]
> Please actually play xeno before commenting on things you have no idea about

Will do.

**Comment by Diegoflores31** [comment]
Played on testmerge . would be interesting if it worked with Disarm too if possible since i was expecting it to work like that.
an audio or something that shows a lil clearly that you are actually extinguishing fire would be really neat

**Comment by Shad0vvs** [comment]
> Played on testmerge . would be interesting if it worked with Disarm too if possible since i was expecting it to work like that. an audio or something that shows a lil clearly that you are actually extinguishing fire would be really neat

Will look into trying to get an audio clip of something, tried to keep it on help to be consistent with patting other players out and just incase you were trying to disarm someone on on fire tile to not give you a cooldown if you click the tile instead.

**Comment by Paperbell** [comment]
> why are xenos getting buffs to make them entierly self-sufficient, yet marines are expected to run everything through a chain of interactions.

I like it this way, it lets me appreciate getting the correct items or cooperate with people to prepare for stuff. I can clear fire way faster than any xeno if I have a portable fire extinguisher. I can drag people way faster than any xeno if I use a roller bed right. They aren't that self-sufficient anyways, they still need help extinguishing flares and they definitely can't solo charge a group if 3+ marines who remotely know what they are doing.

**Comment by stanalbatross** [comment]
> And another thing, why are xenos getting buffs to make them entierly self-sufficient, yet marines are expected to run everything through a chain of interactions.

https://dictionary.cambridge.org/dictionary/english/asymmetric

**Comment by Blithering** [comment]
From playing the testmerge, makes HIDP grenades and anything where you're not constantly reapplying fire near useless at area denial, buys about two seconds. Also runners high attack speed means they can clear flames far faster, clearing white fire at about the speed a carrier was clearing yellow.

**Comment by gohdatrice** [comment]
I noticed on Shiva I couldn't pat certain fire, I think it was only when it was on icy dirt. Could it be due to the code that handles digging snow?

File: code/_onclick/xeno.dm [hunk 1]
```diff
@@ -37,12 +37,34 @@
...
+										FF.update_flame()
+									else qdel(FF)
+								else qdel(FF)
 				xeno_miss_delay(src)
 				animation_attack_on(target)
 				playsound(loc, 'sound/weapons/alien_claw_swipe.ogg', 10, 1) //Quiet to limit spam/nuisance.
-				visible_message(SPAN_DANGER("[src] swipes at [target]!"), \
-				SPAN_DANGER("You swipe at [target]!"), null, 5, CHAT_TYPE_XENO_COMBAT)
+				if(firepatted)
+					visible_message(SPAN_DANGER("[src] pats at the fire!"), \
```
Comment (carlarctg) [inline code/_onclick/xeno.dm:63]: these don't show up when patting fire, pretty sure you're missing this below but i may be wrong
```suggestion
					src.visible_message(SPAN_DANGER("[src] pats at the fire!"), \
```

File: code/_onclick/xeno.dm [hunk 2]
```diff
@@ -37,12 +37,34 @@
...
+					SPAN_DANGER("You pat the fire!"), null, 5, CHAT_TYPE_XENO_COMBAT)
+				else
+					visible_message(SPAN_DANGER("[src] swipes at [target]!"), \
```
Comment (carlarctg) [inline code/_onclick/xeno.dm:66]: ditto
```suggestion
					src.visible_message(SPAN_DANGER("[src] swipes at [target]!"), \
```

File: code/_onclick/xeno.dm [hunk 3]
```diff
@@ -37,12 +37,34 @@
 		if(XENO_NO_DELAY_ACTION)
 			next_move = world.time
 		else
-			if(!tile_attack)
+			if(!tile_attack) // Patting flames for Xenos
+				var/firepatted = FALSE
+				if(src.a_intent == INTENT_HELP)
+					var/fire_level_to_extinguish = 5
+					var/turf/T = target
```
Comment (morrowwolf) [inline code/_onclick/xeno.dm:44]: Not sure if target is guaranteed to be a turf at this point. Also T should be named better.

File: code/_onclick/xeno.dm [hunk 4]
```diff
@@ -37,12 +37,34 @@
...
+					for(var/obj/flamer_fire/FF in T)
```
Comment (morrowwolf) [inline code/_onclick/xeno.dm:45]: FF should be named better

File: code/_onclick/xeno.dm [hunk 5]
```diff
@@ -37,12 +37,34 @@
...
+										FF.update_flame()
+									else qdel(FF)
+								else qdel(FF)
 				xeno_miss_delay(src)
 				animation_attack_on(target)
 				playsound(loc, 'sound/weapons/alien_claw_swipe.ogg', 10, 1) //Quiet to limit spam/nuisance.
-				visible_message(SPAN_DANGER("[src] swipes at [target]!"), \
-				SPAN_DANGER("You swipe at [target]!"), null, 5, CHAT_TYPE_XENO_COMBAT)
+				if(firepatted)
+					src.visible_message(SPAN_DANGER("[src] pats at the fire!"), \
```
Comment (morrowwolf) [inline code/_onclick/xeno.dm:63]: \The [src]

File: code/_onclick/xeno.dm [hunk 6]
```diff
@@ -37,12 +37,34 @@
...
+					SPAN_DANGER("You pat the fire!"), null, 5, CHAT_TYPE_XENO_COMBAT)
+				else
+					src.visible_message(SPAN_DANGER("[src] swipes at [target]!"), \
```
Comment (morrowwolf) [inline code/_onclick/xeno.dm:66]: \The [src] and \the [target]

File: code/_onclick/xeno.dm [hunk 7]
```diff
@@ -37,12 +37,34 @@
...
+					SPAN_DANGER("You swipe at [target]!"), null, 5, CHAT_TYPE_XENO_COMBAT)
```
Comment (morrowwolf) [inline code/_onclick/xeno.dm:67]: \the [target]

**Review by morrowwolf — CHANGES_REQUESTED** [review]

**Comment by calvindoyl** [comment]
Would like to shed a tear for this excellent idea and implementation. Was a great balance idea, made it so fire wasn't perma no mans land unless you had an extinguisher.

Every time I'm stuck staring at the fire for a minute that a dual-wielding marine leaves to guard his retreat while dragging multiple dead corpses back, I try patting.

Not because I think it will work, but just for the memories of when it did.

File: code/_onclick/xeno.dm (hunk 3 above)
Comment (Shad0vvs) [reply code/_onclick/xeno.dm:44]: It has to be because we're in the isturf(target) here.
